modules/thermo-cycler/thermo-cycler-arduino/lights.cpp

```diff
@@ -0,0 +1,165 @@
+#include "lights.h"
+
+#if RGBW_NEO
+Adafruit_NeoPixel_ZeroDMA strip(NUM_PIXELS, NEO_PIN, NEO_GRBW);
+#else
+Adafruit_NeoPixel_ZeroDMA strip(NUM_PIXELS, NEO_PIN, NEO_GRB + NEO_KHZ800);
+#endif
+
+Lights::Lights(){}
+
+void Lights::setup()
+{
+  pinMode(NEO_PWR, OUTPUT);
+  pinMode(NEO_PIN, OUTPUT);
+  digitalWrite(NEO_PWR, HIGH);
+  strip.begin();
+  strip.setBrightness(70);
+  _status = Light_status::solid;
+  _color = Light_color::white;
+  strip.show();
+}
+
+void Lights::update()
+{
+  // Updates lights acc to _status and _color
+  switch(_status)
+  {
+    case Light_status::solid:
+      _set_strip_color(static_cast<uint32_t>(_color));
+      break;
+    case Light_status::pulsing:
+      _pulse_leds(_color);
+      break;
+    case Light_status::wipe:
+      _color_wipe(Light_color::none, 50);
+      _color_wipe(_color, 50);
+      break;
+    case Light_status::all_off:
+    default:
+      _set_strip_color(static_cast<uint32_t>(Light_color::none));
+      break;
+  }
+}
+
+void Lights::set_lights(TC_status tc_status)
+{
+  // Set _status and _color acc to tc_status
+  switch(tc_status)
+  {
+    case TC_status::idle:
+      _status = Light_status::solid;
+      _color = Light_color::white;
+      break;
+    case TC_status::going_to_hot_target:
+      _status = Light_status::pulsing;
+      _color = Light_color::red;
+      break;
+    case TC_status::going_to_cold_target:
+      _status = Light_status::pulsing;
+      _color = Light_color::blue;
+      break;
+    case TC_status::at_hot_target:
+      _status = Light_status::solid;
+      _color = Light_color::red;
+      break;
+    case TC_status::at_cold_target:
+      _status = Light_status::solid;
+      _color = Light_color::blue;
+      break;
+    case TC_status::in_cycle_hot_target:
+      _status = Light_status::wipe;
+      _color = Light_color::red;
+      break;
+    case TC_status::in_cycle_cold_target:
+      _status = Light_status::wipe;
+      _color = Light_color::blue;
+      break;
+    default:
+      _status = Light_status::solid;
+      _color = Light_color::orange;
+      break;
+  }
+}
+
+void Lights::set_lights(Light_status status, Light_color color)
+{
+  // Set _status and _color to args values
+  _status = status;
+  _color = color;
+}
+
+void Lights::_set_strip_color(uint32_t color)
+{
+  for(int i=0; i<strip.numPixels(); i++)
+  {
+    strip.setPixelColor(i, color); // strip.setPixelColor(n, [green, red, blue, white]);
+    strip.show();
+    delay(1);
+  }
+}
+
+void Lights::_color_wipe(Light_color color, int speed_delay) {
+  for(uint16_t i=0; i<strip.numPixels(); i++)
+  {
+    _set_pixel(i, color);
```

Comment:
Do we want similar state management stuff here as we do in pulse so we're not delaying a couple hundred ms every time we call this?